thunder/tests/distributed/test_fsdp.py

```diff
@@ -82,17 +82,26 @@ def func(
         b = make_tensor((2, 2), device=device, dtype=torch.float32)
         process_group = c10d.new_group()
         _ = cfunc(a, b, process_group)
-        execution_trace = thunder.last_traces(cfunc)[-2]
-        sorted_execution_trace = sort_waits(execution_trace)
+        traces = thunder.last_traces(cfunc)
+
+        del_last_used_indices = [
+            i for i, trace in enumerate(traces) if "Delete Last Used" in traces.get_provenance().pss
```

Comment:
`last_traces(cfunc)[-2]` used to be the trace just before `del_last_used` is applied. Now we have `Unwrap the return value` trace after `del_last_used`, so indexing by -2 no longer works.

Spelling out `"Delete Last Used"` is not very nice, but this makes the intention easily recognizable.